csrc/index_compute.cpp

```diff
@@ -2655,9 +2651,27 @@ std::pair<Val*, Val*> Index::getCpAsyncBulkGmemIndex(
 
   ValGroups groups_to_index = tma_info.getTMADomain();
 
+  // TensorIndexer needs IterDomain instead of ValGroup to work around
+  // the resize indexing issue
+  std::vector<IterDomain*> ids_to_index;
+  ids_to_index.reserve(groups_to_index.size());
+  const auto tma_all_ids = is_load ? consumer_tv->domain()->allIDs()
+                                   : producer_tv->domain()->allIDs();
+  for (const auto& group : groups_to_index) {
+    auto it = std::find_if(
+        tma_all_ids.begin(), tma_all_ids.end(), [&](IterDomain* gmem_id) {
+          return group->has(gmem_id);
+        });
+    NVF_ERROR(
+        it != tma_all_ids.end(),
+        "Cannot find corresponding ID for ",
+        nvfuser::toString(group));
+    ids_to_index.push_back(*it);
+  }
```

Comment:
Does the unit tests of TMA all pass? I remembered that some ids may be newly created IDs, therefore not in the `allIDs` of the tv.

Reply:
That's I wanted to confirm with the CI tests. I'll check them manually.

Reply:
OK, these tests failed here:

```
[  FAILED  ] TMAIndexingTest.DefineBoxByCompositing1
[  FAILED  ] TMAIndexingTest.DefineBoxByCompositing2
[  FAILED  ] TMAIndexingTest.NonTrivialGmemAllocationDomain2
```

I think we can just pick any ID from the ValGroup. The arguments to TensorIndexer doesn't need to be those of the indexed tensors. It is required for the resize WAR case (https://github.com/NVIDIA/Fuser/blob/resize_indexing_cyclic_graph_war/csrc/id_model/indexing_traversal.cpp#L81-L104), but as long as TMA is not used resize, it wouldn't fail. 

This is obviously far from ideal since it would fail if both TMA and resize are used but at least that should be a hard failure rather than silently using invalid indexing paths.

Reply:
As a WAR, I am OK with switching to use ID, and using an arbitrary ID sounds good to me. What I am more interested is, in the future, assuming that we have a real fix for the resize problem. Would we still use ID, or will we use ValGroup?

Reply:
I think we should switch back to `ValGroup`, although `Val` may not be just `IterDomain` anymore, like you suggested, we may need to include Expr as a key as well.

Reply:
Confirmed all tests pass on H100.